pkg/kube/controllers/webhook_configuration.go

```diff
@@ -169,26 +175,31 @@ func (f *WebhookConfig) generateValidationWebhookServerConfig(ctx context.Contex
 	for _, webhook := range webhooks {
 		ctxlog.Debugf(ctx, "Calculating validation webhook '%s'", webhook.Name)
 
-		url := url.URL{
-			Scheme: "https",
-			Host:   net.JoinHostPort(f.config.WebhookServerHost, strconv.Itoa(int(f.config.WebhookServerPort))),
-			Path:   webhook.Path,
-		}
-		urlString := url.String()
-		wh := admissionregistrationv1beta1.Webhook{
-			Name:              webhook.Name,
-			Rules:             webhook.Rules,
-			FailurePolicy:     &webhook.FailurePolicy,
-			NamespaceSelector: webhook.NamespaceSelector,
-			ClientConfig: admissionregistrationv1beta1.WebhookClientConfig{
+		if f.config.Provider == "gke" {
+			clientConfig := admissionregistrationv1beta1.WebhookClientConfig{
```

Comment:
I suspect you might need to do the same for EKS (we saw the same problem on EKS). Deleting the webhook made it work so under the hood seems like the EKS apiserver also cannot contact the webhook service using the clusterip/port.

Reply:
Having provider-specific logic hard-coded in the Go code seems pretty inflexible to me and creates a hard dependency between operator release and provider version.

I think it would be better if the different logic would be configured via some property, and that property could be overridden at deployment time. So it could be determined (if possible) by helm, or otherwise is specified by the deployer.

Anyways, the feature modification should be triggered by some underlying feature name and not by a provider name.

Reply:
Is there any reason why we should not use the `service` field in all cases and not special case it? The kube documentation seems to suggest so unless I'm mis-reading it?

https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#url

**The host should not refer to a service running in the cluster**; use a service reference by specifying the service field instead.